lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -302,6 +302,33 @@ static bool recursiveType(const StructType *ST, const Type *Ty) {
   return Run(Ty);
 }
 
+// Add decoration if needed
+void addFPBuiltinDecoration(SPIRVModule *BM, Instruction *Inst,
+                            SPIRVInstruction *I) {
+  const bool AllowFPMaxError =
+      BM->isAllowedToUseExtension(ExtensionID::SPV_INTEL_fp_max_error);
+  if (!AllowFPMaxError)
+    return;
+  if (auto *II = dyn_cast_or_null<IntrinsicInst>(Inst);
+      II && II->getCalledFunction()->getName().startswith("llvm.fpbuiltin")) {
```

Comment:
Can we add a test case for such builtins?

Also such syntax construction is not used in the repo, I find it a bit less obvious than usage of `if (II = dyn_cast())`. Do you mind to change the style here?

Reply:
Thanks for the review, We need to use II later in the loop but let me change the style.

Reply:
BTW we already have tests for the builtins here: `test/extensions/INTEL/SPV_INTEL_fp_max_error/IntelFPMaxError.ll`